src/SignatureBES.cpp

```diff
@@ -1065,76 +1090,92 @@ void SignatureBES::saveToXml(ostream &os) const
     const SignedSignaturePropertiesType::SignatureProductionPlaceOptional& sigProdPlaceOptional =
         getSignedSignatureProperties().signatureProductionPlace();
     if(sigProdPlaceOptional.present() && sigProdPlaceOptional->postalCode().present())
         return sigProdPlaceOptional->postalCode().get();
-    return "";
+    const SignedSignaturePropertiesType::SignatureProductionPlaceV2Optional &sigProdPlaceV2Optional =
+            getSignedSignatureProperties().signatureProductionPlaceV2();
+    if(sigProdPlaceV2Optional.present() && sigProdPlaceV2Optional->postalCode().present())
+        return sigProdPlaceV2Optional->postalCode().get();
+    return string();
 }
 
 string SignatureBES::countryName() const
 {
-    const SignedSignaturePropertiesType::SignatureProductionPlaceOptional& sigProdPlaceOptional =
+    // return elements from SignatureProductionPlace element or SignatureProductionPlaceV2 when available
+    const SignedSignaturePropertiesType::SignatureProductionPlaceOptional &sigProdPlaceOptional =
         getSignedSignatureProperties().signatureProductionPlace();
     if(sigProdPlaceOptional.present() && sigProdPlaceOptional->countryName().present())
         return sigProdPlaceOptional->countryName().get();
-    return "";
+    const SignedSignaturePropertiesType::SignatureProductionPlaceV2Optional &sigProdPlaceV2Optional =
+            getSignedSignatureProperties().signatureProductionPlaceV2();
+    if(sigProdPlaceV2Optional.present() && sigProdPlaceV2Optional->countryName().present())
+        return sigProdPlaceV2Optional->countryName().get();
+    return string();
 }
 
-/**
- * The role that signer claims to hold while signing.
- *
- * @return returns the claimed role of the signer.
- */
 vector<string> SignatureBES::signerRoles() const
 {
     vector<string> roles;
-    const SignedSignaturePropertiesType::SignerRoleOptional& roleOpt =
+    const SignedSignaturePropertiesType::SignerRoleOptional &roleOpt =
         getSignedSignatureProperties().signerRole();
-    if ( !roleOpt.present() )
-        return roles;
-
-    const SignerRoleType::ClaimedRolesOptional& claimedRoleOpt = roleOpt->claimedRoles();
-    if ( !claimedRoleOpt.present() )
-        return roles;
-
-    for(const xades::AnyType &type: claimedRoleOpt->claimedRole())
+    const SignedSignaturePropertiesType::SignerRoleV2Optional &roleV2Opt =
+        getSignedSignatureProperties().signerRoleV2();
+    for(const ClaimedRolesListType::ClaimedRoleType &type: [&]() ->
+        ClaimedRolesListType::ClaimedRoleSequence const {
+        // return elements from SignerRole element or SignerRoleV2 when available
+        if(roleOpt.present())
+            return roleOpt->claimedRoles()->claimedRole();
+        else if(roleV2Opt.present())
+            return roleV2Opt->claimedRoles()->claimedRole();
+        else
+            return ClaimedRolesListType::ClaimedRoleSequence();
+    }())
```

Comment:
The indentation here is a bit misleading, at first glance the code above could be seen as the for-loop body.
I don't think we actually have a standard for indenting lambdas, but it might help to push it further right and add a set of curly-braces around the line below